ui/permissions.py

```diff
@@ -68,4 +204,7 @@ def has_permission(self, request, view):
             uuid.UUID(collection_key)
         except ValueError:
             raise ValidationError('wrong UUID format for {}'.format(collection_key))
-        return Collection.objects.filter(owner=request.user, key=collection_key).exists()
+        collection = Collection.objects.filter(key=collection_key)
+        if collection:
+            return has_admin_permission(collection.first(), request)
+        return False
```

Comment:
Right. I'm saying that we can write the same thing more succinctly (no `if` clause needed) with no extra queries/logic involved. Unless I'm mistaken, these two bits of code are functionally identical with no computational difference:

```python
collection = Collection.objects.filter(key=collection_key).first()
return collection is not None and has_admin_permission(collection, request)
```

```python
collection = Collection.objects.filter(key=collection_key)
if collection:
    return has_admin_permission(collection.first(), request)
return False
```